x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/TokenAuthIntegTests.java

```diff
@@ -828,6 +837,62 @@ public void testMetadataIsNotSentToClient() {
         assertFalse(clusterStateResponse.getState().customs().containsKey(TokenMetadata.TYPE));
     }
 
+    public void testCreatorRealmCaptureWillWorkWithClientRunAs() throws IOException {
+        final String nativeTokenUsername = "native_token_user";
+        getSecurityClient().putUser(new User(nativeTokenUsername, "superuser"), SecuritySettingsSourceField.TEST_PASSWORD_SECURE_STRING);
+
+        final Client runAsClient = client().filterWithHeader(
+            Map.of(
+                "Authorization",
+                UsernamePasswordToken.basicAuthHeaderValue(ES_TEST_ROOT_USER, TEST_PASSWORD_SECURE_STRING.clone()),
+                AuthenticationServiceField.RUN_AS_USER_HEADER,
+                nativeTokenUsername
+            )
+        );
+
+        // Create a token with client credentials and run-as request header
+        var createTokenRequest1 = new org.elasticsearch.xpack.core.security.action.token.CreateTokenRequest();
+        createTokenRequest1.setGrantType("client_credentials");
+        final PlainActionFuture<org.elasticsearch.xpack.core.security.action.token.CreateTokenResponse> future1 = new PlainActionFuture<>();
+        runAsClient.execute(CreateTokenAction.INSTANCE, createTokenRequest1, future1);
+        final String accessToken = future1.actionGet().getTokenString();
+        var invalidateTokenRequest = new org.elasticsearch.xpack.core.security.action.token.InvalidateTokenRequest(
+            null,
+            null,
+            "index",
+            nativeTokenUsername
+        );
```

Comment:
Should we authenticate with the token here?
The test is checking that the token is created correctly with the right realm, and it feels like the best way to check both those things (token works, token is for correct user/realm) is via `authenticate`.

Reply:
I added authentication. It is nice to have. But checking authentication itself does not do what I need. The intention is to test the fix of realm capture, which is only used when invalidating or refreshing tokens.